### DISCUSSION TITLE
wifi for ever in WL_NO_SHIELD

### DISCUSSION THREAD

**Discussion by JosVanPutte** [AK-Homberger/NMEA2000-Workshop#5] [Q&A]
Hi,
I bought the ESP32 dev module from AZ delivery and it fails to connect to the WiFi. The status is WL_NO_SHIELD and that seems to indicate that I have the wrong board selected. The forums report that this status inidicates that the WiFi module is not found ?

The Additional boards manager URLs is :
https://dl.espressif.com/dl/package_esp32_index.json,
https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_dev_index.json,
https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json

I use the Arduino IDE (2.0.3) with board 'ESP32 Dev Module'. The pinout of the dev module matches the description in the WiFi AIS gateway. I have tried a number of ESP32 boards but they all give me the same status.

There seem to be different URLs for the ESP32 boards but none of them list the (AZDelivery) ESP32 dev module V2.

**Comment by AK-Homberger** [comment]
As stated in the preparation section the Workshop is tested with V1 of the ESP32 Development Module and Board version 1.0.6. For the ESP32 V2 and the newer Board versions (2.x.x) there might be additional changes necessary (different URL and code changes). Unfortunately I, don't own a V2 ESP32 to test it.

Timo did also alot of changes to the NMES 2000 library. For the workshop it might be necessry to use the library version used for the workshop. I added to old libraries to GitHub under "Libraries".

**Reply by JosVanPutte** [reply]
Thanks for the quick reply, at least then I know the selection of the ESP Dev Module is correct. I'll try to get in contact with AZDelivery, maybe they have more information..
In the mean time, I'll play around with different configurations to see if I can get the WiFi to work.

I started this project because I added a new display to my boat (Elements 7) for AIS perimiter alarms when I am alone inside. It was incompatible with my old MFD outside (E7D) and refused to work when connecting both to the same STng bus ([here](https://youtu.be/gh9qbfKd_6U)). 

I fixed it by using two STng buses, and it does what I need.. But at Boot Dusseldorf I saw the stand of [open boat projects](https://open-boat-projects.org/de/).
Fantastic ! It will keep me busy until I can go sailing again.. :-) (I saw a pickture of a Bavaria sailing yacht in this project, I have a Cruiser 36..)

As my boat is still on land and I like a challenge, I started to use a YakBitz ST1 to NMEA0183 converter and use your Wind2Nk2000 project to pass the wind (and add the depth..) to the N2K bus of the display inside.

As the converter has not arrived yet, and I noticed at Aisler that I would get 3 PCBs, I thought it was also nice to have the AIS data on Wifi for my tablet. Also the PCBs have not arrived yet, so I am just playing around with the dev module connected to my PC.

For the 3rd board, maybe it would be nice to implement the thermostat interface to the ITTT of my Google Nest, so I can say 'Hey Google, set course to 15 degrees'. In fact I tested it by renaming my heater to course, and google actually replied with 'I will set course to 15 degrees'.. Hilarious.

**Reply by JosVanPutte** [reply]
I found the problem. I have the Dev Kit, not the Dev Module.. With the DOIT ESP32 DEVKIT V1 the wifi works !

**Reply by JosVanPutte** [reply]
It now works with the Dev Module too.
The problem seemed to be in the resetWifiSettings nvram clearing because when I set the flag to false, the WiFi works well.
